api/v2/paths/directory.js

```diff
@@ -0,0 +1,28 @@
+const appRoot = require('app-root-path');
+
+const directoryDao = require('../db/ldap/directory-dao');
+
+const { errorHandler, errorBuilder } = appRoot.require('errors/errors');
+const { openapi: { paths } } = appRoot.require('utils/load-openapi');
+
+/**
+ * @summary Get directory
+ */
+const get = async (req, res) => {
+  try {
+    const result = await directoryDao.getDirectories(req.query);
+    return res.send(result);
+  } catch (err) {
+    if ('lde_message' in err) {
+      if (err.lde_message.includes('Size Limit Exceeded')) {
+        return errorBuilder(res, 400, ['Size Limit of 200 Results Exceeded (search too broad)']);
+      }
+      return errorHandler(res, [err.lde_message]);
```

Comment:
Why `err.lde_message` need to be wrapped within an array?

Reply:
the skeleton expects a list of error messages in general, so if a single string is passed it is interpreted as a list of characters:
```
{
    "errors": [
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "S",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "i",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "z",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "e",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": " ",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "L",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "i",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "m",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "i",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "t",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": " ",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "o",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "f",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": " ",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "2",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "0",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "0",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": " ",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "R",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "e",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "s",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "u",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "l",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "t",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "s",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": " ",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "E",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "x",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "c",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "e",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "e",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "d",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "e",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "d",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": " ",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "(",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "s",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "e",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "a",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "r",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "c",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "h",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": " ",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "t",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "o",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "o",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": " ",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "b",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "r",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "o",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "a",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": "d",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        },
        {
            "status": "400",
            "code": "1400",
            "title": "Bad Request",
            "detail": ")",
            "links": {
                "about": "https://developer.oregonstate.edu/documentation/error-reference#1400"
            }
        }
    ]
}
```